pkg/reconciler/openshift/openshift_test.go

```diff
@@ -107,3 +107,14 @@ func TestAllowedNamespaces(t *testing.T) {
 	clusterConfigNamespaces = "foo,bar"
 	assert.DeepEqual(t, false, allowedNamespace(argocdNamespace, clusterConfigNamespaces))
 }
+
+func TestReconcileArgoCD_reconcileRedisDeployment(t *testing.T) {
+	a := makeTestArgoCD()
+	testDeployment := makeTestDeployment()
+	testDeployment.ObjectMeta.Name = a.Name + "-" + "redis"
+
+	want := append([]string{"redis-server"}, testDeployment.Spec.Template.Spec.Containers[0].Args ...)
+
+	assert.NilError(t, reconcilerHook(a, testDeployment))
+	assert.DeepEqual(t, testDeployment.Spec.Template.Spec.Containers[0].Args, want)
+}
```

Comment:
Can you add more tests here ?

Reply:
added another test case to verify that deployment hook is only executed for redis deployment.